src/QMCDrivers/WalkerControlBase.cpp

```diff
@@ -385,11 +385,23 @@ int WalkerControlBase::sortWalkers(MCWalkerConfiguration& W)
   //W.EnsembleProperty.Variance=(e2sum*wsum-esum*esum)/(wsum*wsum-w2sum);
   if (!WriteRN)
   {
-    if(good_w.empty())
+    // single rank run, need at least one good walker
+    if( NumContexts==1 && good_w.empty() )
```

Comment:
Please merge the two if statements ( if (a || b ) {} ) so that the error message is not duplicated. Then update the error "All the walkers have died. Improve the trial wavefunction or adjust the simulation parameters". We don't need to be contacted.